src/atlas/domain/sequence_diagram.clj

```diff
@@ -96,5 +114,19 @@
 
 (s/defn arrows :- [s-sequence-diagram/Arrow]
   [trace :- s-jaeger/Trace]
-  (let [client-spans (->> trace :spans (filter client-span?))]
-    (reduce (span->arrow-pair trace) [] client-spans)))
+  (let [client-span-arrows (->> trace
```

Comment:
instead of calling a reducer for producer-arrows and consumer-arrows you could just use a map